pkg/internal/codegen/parse/util_test.go

```diff
@@ -123,3 +124,63 @@ func TestParseScaleParams(t *testing.T) {
 		}
 	}
 }
+func Test_parsePrintColumnParams(t *testing.T) {
+	tests := []struct {
+		name     string
+		tag      string
+		expected []v1beta1.CustomResourceColumnDefinition
+		parseErr error
+	}{
+		{
+			name: "test ok",
+			tag:  "+kubebuilder:printcolumn:name=toy,type=string,JSONPath=.status.conditions[?(@.type==\"Ready\")].status,description=descr1,priority=3,format=date",
+			expected: []v1beta1.CustomResourceColumnDefinition{
+				{
+					Name:        "toy",
+					Type:        "string",
+					Format:      "date",
+					Description: "descr1",
+					Priority:    3,
+					JSONPath:    ".status.conditions[?(@.type==\"Ready\")].status",
+				},
+			},
+			parseErr: nil,
+		},
+		{
+			name: "Minimum Three parameters",
+			tag:  "+kubebuilder:printcolumn:name=toy,type=string,JSONPath=.status.conditions[?(@.type==\"Ready\")].status",
+			expected: []v1beta1.CustomResourceColumnDefinition{
+				{
+					Name:     "toy",
+					Type:     "string",
+					JSONPath: ".status.conditions[?(@.type==\"Ready\")].status",
+				},
+			},
+			parseErr: nil,
+		},
+		{
+			name:     "two parameters",
+			tag:      "+kubebuilder:printcolumn:name=toy,type=string",
+			expected: []v1beta1.CustomResourceColumnDefinition{},
+			parseErr: fmt.Errorf(printColumnError),
+		},
+		{
+			name:     "one requied parameter is missing",
+			tag:      "+kubebuilder:printcolumn:name=toy,type=string,description=.status.conditions[?(@.type==\"Ready\")].status",
+			expected: []v1beta1.CustomResourceColumnDefinition{},
+			parseErr: fmt.Errorf(printColumnError),
+		},
```

Comment:
Thanks for adding unit-test here. We can add for invalid priority and format as well.

Reply:
Ack. Done!